opendevin/runtime/runtime.py

```diff
@@ -143,9 +142,14 @@ async def add_env_vars(self, env_vars: dict[str, str]) -> None:
 
     async def on_event(self, event: Event) -> None:
         if isinstance(event, Action):
+            # set timeout to default if not set
+            if event.timeout is None:
+                event.timeout = self.config.sandbox.timeout
```

Comment:
@enyst I removed the additional argument for the CmdRun and IPython, made `.timeout` a special attribute like `_cause`, and updated it if the agent developer did not manually specify a value.